### DISCUSSION TITLE
Is a velero client deprecated?

### DISCUSSION THREAD

**Discussion by dev-a-z** [vmware-tanzu/velero#7971] [Community support Q&A] [answered]
I bumped the velero version from `v1.9.5` to `v1.14.0` and found out a velero client was deprecated in [factory.go](https://github.com/vmware-tanzu/velero/blob/4f5a61b8e7330367fc851e8c8f8c3a89120e342c/pkg/client/factory.go#L44)

then, Is there not supported a `velero client` anymore? if yes, should I choose a `kube client` or `kubebuilder client`?
Thanks!

**Comment by blackpiglet** [comment, marked as answer]
Yes. The generated Veleo clients are deprecated, and they will be removed in v1.15.
Sure, you can try will client-go, kubebuilder client, or the controller-runtime client.

**Reply by dev-a-z** [reply]
Got it! Thanks for the answer!
